opm/utility/ECLResultData.hpp

```diff
@@ -0,0 +1,178 @@
+        explicit ECLResultData(const Path& casePrefix);
+
+        /// Copy constructor.
+        ///
+        /// \param[in] rhs Object from which to construct new instance.
+        ECLResultData(const ECLResultData& rhs);
+
+        /// Move constructor.
+        ///
+        /// \param[in,out] rhs Object from which to construct new instance.
+        ///    Its internal implementation will be subsumed into the new
+        ///    object.
+        ECLResultData(ECLResultData&& rhs);
+
+        /// Assignment operator.
+        ///
+        /// \param[in] rhs Object from which to assign new values to current
+        ///    instance.
+        ///
+        /// \return \code *this \endcode.
+        ECLResultData& operator=(const ECLResultData& rhs);
+
+        /// Move assignment operator.
+        ///
+        /// \param[in,out] Object from which to assign new instance values.
+        ///    Its internal implementation will be subsumed into this
+        ///    instance.
+        ///
+        /// \return \code *this \endcode.
+        ECLResultData& operator=(ECLResultData&& rhs);
+
+        /// Destructor.
+        ~ECLResultData();
+
+        /// Access the underlying ERT representation of the result-set.
+        ///
+        /// This is essentially a hole in the interface that is intended to
+        /// support a few very specialised uses.  Handle with care.
+        ///
+        /// \return Handle to underlying ERT representation of result-set.
+        const ecl_file_type* getRawFilePtr() const;
+
+        /// Reset the object's internal view of the result-set to encompass
+        /// the entirety of the underlying file's result vectors.
+        ///
+        /// This is mostly useful in the case of accessing static result
+        /// sets such as INIT files.
+        ///
+        /// \return Whether or not generating the global view succeeded.
+        bool selectGlobalView();
+
+        /// Select a result-set view that corresponds to a single report
+        /// step.
+        ///
+        /// This is needed when working with dynamic restart data.
+        ///
+        /// \param[in] step Report step number.
+        ///
+        /// \return Whether or not selecting the report step succeeded.  The
+        ///    typical failure case is the report step not being available
+        ///    in the result-set.
+        bool selectReportStep(const int step);
+
+        /// Query current result-set view for availability of particular
+        /// named result vector in particular enumerated grid.
+        ///
+        /// \param[in] vector Named result vector for which to query data
+        ///    availability.
+        ///
+        /// \param[in] gridID Identity of specific grid for which to query
+        ///    data availability.
+        ///
+        /// \return Whether or not keyword data for the named result vector
+        ///    is available in the specific grid.
+        bool haveKeywordData(const std::string& vector,
```

Comment:
We should avoid using variable names that are easy to confuse with standard functions or classes, I don't think the term vector is essential enough here to override that: "field" or "keyword" would be possible alternatives.